pennylane/pauli/pauli_arithmetic.py

```diff
@@ -207,15 +221,48 @@ def to_mat(self, wire_order, format="dense"):
             if wire_order is None or wire_order == wires.Wires([]):
                 raise ValueError("Can't get the matrix of an empty PauliWord.")
             return (
-                np.eye(2 ** len(wire_order))
+                np.diag([coeff] * 2 ** len(wire_order))
                 if format == "dense"
-                else sparse.eye(2 ** len(wire_order), format=format)
+                else coeff * sparse.eye(2 ** len(wire_order), format=format)
             )
 
-        matrix_map = sparse_mat_map if format != "dense" else mat_map
-        kron = sparse.kron if format != "dense" else math.kron
-
-        return reduce(kron, (matrix_map[self[w]] for w in wire_order))
+        wire_order = wire_order or self.wires
+        if format == "dense":
+            return coeff * reduce(math.kron, (mat_map[self[w]] for w in wire_order))
+
+        return self._to_sparse_mat(wire_order, coeff)
+
+    def _to_sparse_mat(self, wire_order, coeff):
+        """Compute the sparse matrix of the Pauli word times a coefficient, given a wire order."""
+        full_word = [self[wire] for wire in wire_order]
+        matrix_size = 2 ** len(wire_order)
+        data = np.empty(matrix_size, dtype=np.complex128)  # Non-zero values
+        indices = np.empty(matrix_size, dtype=np.int64)  # Column index of non-zero values
+        indptr = _cached_arange(matrix_size + 1)  # Non-zero entries by row (starting from 0)
+
+        current_size = 2
+        data[:current_size], indices[:current_size] = _cached_sparse_data(full_word[-1])
+        data[:current_size] *= coeff  # Multiply initial term better than the full matrix
+        for s in full_word[-2::-1]:
+            if s == "I":
```

Comment:
As mentioned above, branching based on strings is usually much slower than integers (unless Python is smart and treats them as `char`s maybe). If profiling shows this is a bottleneck, consider mapping strings to integer first.

Reply:
I opted for this option to enhance code readability. The Pauli word comes naturally in string format, so we do not need to convert it to ints (mapping trough a dictionary), which ends up compensating for this branching afterwards. I did some rough testing and they are within error bars of each other:
14 qubits, 25 Pauli words, strings -> total time $2.49ms\pm 110\mu s$
14 qubits, 25 Pauli words, ints      -> total time $2.65ms\pm 131\mu s$
25 qbuits, 25 Pauli words, strings -> total time $5.51s\pm 141ms$
25 qubits, 25 Pauli words, ints      -> total time $5.49s\pm 102ms$

Reply:
It is rather large operations after all, so it make sense that it doesn't matter. Now we know the answer for sure. Thanks @BorjaRequena